core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -151,6 +157,14 @@ func (r *logRecoverer) Start(pctx context.Context) error {
 }
 
 func (r *logRecoverer) Close() error {
+	if err := r.stop(); err != nil {
```

Comment:
This smells like a logical race/mishandling of where a lock belongs - why do we need to attempt to obtain and then immediately drop the lock here?

Reply:
We use the lock for thread safe access to the cancel func.

Reply:
ah; my bad. misread the diff